File: 2021-07-21/hanoi_split/att/template_sol.cpp
/**
 * Un template per la tua soluzione di hanoi_split
 * Romeo Rizzi, 2021-07-20
 *
 */

#include <cassert>
#include <cstdio>

//const int DEBUG = 0;
const int DEBUG = 1;

typedef long long unsigned int llu;

int t, N;

// implemento prima la simulazione (più facile, mi abbandono agli stili descrittivo e ricorsivo)
void sposta_disco(int n, char peg_from, char peg_to) {
  printf("sposta il disco %d dal piolo %c al piolo %c\n", n, peg_from, peg_to);
}


int main() {
  scanf("%d%d", &t, &N);
  if(t==0)
    printf("%llu\n", 42); // TO DO: stampa il numero di mosse
  else {
    // TO DO (il template si limita a fornire la lista esplicita di mosse per N=3. A tè introdurre le procedure con la giusta logica in questo file
    sposta_disco(1,'A','B');
    sposta_disco(2,'A','C');
    sposta_disco(1,'B','C');
    sposta_disco(3,'A','B');
    sposta_disco(1,'C','B');
  }
        
  return 0;
}

File: 2021-07-21/hanoi_split/att/template_sol.py
#!/usr/bin/env python3
# -*- coding: latin-1 -*-
"""Un template per la tua soluzione di hanoi_split
   Romeo Rizzi, 2021-06-20
"""
import sys
sys.setrecursionlimit(1000000)

t, N = map(int,input().split()) # problem type and number of disks
assert 0 <= t <= 1

def sposta_disco(i, peg_from, peg_to):
    print(f"sposta il disco {i} dal piolo {peg_from} al piolo {peg_to}")
    
if t == 0:
    print(42) # TO DO: stampa tù il numero corretto di mosse, ovviamente il template non lo fa per te
else:
    # TO DO (il template si limita a fornire la lista esplicita di mosse per N=3. A tè introdurre le procedure con la giusta logica in questo file
    sposta_disco(1,'A','B')
    sposta_disco(2,'A','C')
    sposta_disco(1,'B','C')
    sposta_disco(3,'A','B')
    sposta_disco(1,'C','B')
